DEVELOPING.md

```diff
@@ -8,6 +8,16 @@ Before running through the release it's good to run the build and the tests loca
 also test-drive the commit in an existing Bazel workspace to sanity check functionality.
 
 #### Steps
+1. (Optional) modify the `./python/pip_install/tools/requirements.txt` file and run `./tools/update_pip_deps.sh` script:
```

Comment:
Lets move this to another section that is focused on  updating internal tool dependencies. These aren't really release steps; it's a chore that should be done regularly but also needs verification.